### PR TITLE
tfmini : fix scheduling, parsing and modernize output

### LINKED ISSUE #10264: Tfmini rangefinder detection have huge delay when use the telem1 as a radio link
**Describe the bug**
**Tfmini rangefinder detection have huge delay when use  the telem1 is enable as a radio link .**

When using the Tfmini rangefinder at telem2 with default setting and using radio link in telem1 at the same time, the rangefinder will not work well, and the range detection will drop to from a very high rate to about 2s per detection. This can be directly view from QGC Analyze.

**To Reproduce**
1. Start the pixhawk4
2.plug the radio link in telem1 and tfmini in telem2
3. enable tfmini in param setting
4. use the radio link and QGC analyze can see a huge delay of range detection.
5.unplug the radio link the detection rate will back to normal.

**Expected behavior**
The tfmini should work in about 100hz rate.

**Log Files and Screenshots**

**Drone (please complete the following information):**
- Pixhawk4 with 1.8.0 PX4 FIRMWARE.

### PR DESCRIPTION
**Pull request by mhkabir**
This PR :
1. Reduces the driver scheduling interval to 9ms, such that the driver is always ready to read new data. Running it at exactly 100Hz is not correct since the driver and sensor measurement intervals are not "in sync", causing the driver to miss data. This caused a fill-up of the UART buffer.
2. Rejigs parsing logic to always publish the latest measurement. This gets rid of the 600ms delay we had been seeing, since the driver was always parsing old data from the UART buffer, fixed at 10 bytes per every collect call.
3. Removes useless reset() function.
4. Modernizes output, removing `errx()` and `warnx()` calls.

Fixes #10264

### CODE REVIEW

**Comment by mhkabir** [comment]
@Antiheavy @eleboss

**Comment by eleboss** [comment]
Cool！ I will try that tomorrow.

**Comment by mhkabir** [comment]
Actually, I just checked and this doesn't seem to have fixed the issue. The 600ms lag is still there, even though the sensor is publishing at full rate now.

**Comment by mhkabir** [comment]
I tried flushing the buffer after every successful parse, but the data is still delayed. I'm beginning to suspect that this is a sensor issue.

**Comment by eleboss** [comment]
Let me try to directly ask the benewake.

**Comment by dagar** [comment]
CI failure is astyle.

File: src/drivers/distance_sensor/tfmini/tfmini.cpp [hunk 1]
```diff
@@ -730,7 +715,8 @@ start(const char *port, uint8_t rotation)
 	int fd;
 
 	if (g_dev != nullptr) {
-		errx(1, "already started");
+		PX4_ERR("already started");
+		exit(1);
```
Comment (dagar) [inline src/drivers/distance_sensor/tfmini/tfmini.cpp]: Can you drop the exits while you're at it? This is otherwise close fo functional on a raspberry pi.

**Comment by eleboss** [comment]
Tested, works fine!

Real-time makes the world better, thank you.

**Comment by LorenzMeier** [comment]
I rebased and addressed the comments from @dagar.

**Comment by eleboss** [comment]
It seems this modification can't compile on @dagar release/1.8 branch.

**Comment by mhkabir** [comment]
Cygwin doesn't like this :

```
../../src/drivers/distance_sensor/tfmini/tfmini.cpp: In member function ‘int TFMINI::collect()’:
../../src/drivers/distance_sensor/tfmini/tfmini.cpp:535:15: error: ‘FIONREAD’ was not declared in this scope
  ::ioctl(_fd, FIONREAD, (unsigned long)&bytes_available);
               ^~~~~~~~
compilation terminated due to -Wfatal-errors.
```

File: src/drivers/distance_sensor/tfmini/tfmini.cpp [hunk 2]
```diff
@@ -534,43 +531,56 @@ TFMINI::collect()
...
-			valid = true;
+		/* parse buffer */
+		for (int i = 0; i < ret; i++) {
+			tfmini_parse(readbuf[i], _linebuf, &_linebuf_index, &_parse_state, &distance_m);
 		}
-	}
 
-	if (!valid) {
+		/* bytes left to parse */
+		bytes_available -= ret;
```
Comment (bkueng) [inline src/drivers/distance_sensor/tfmini/tfmini.cpp]: If for some reason `read` returns 0, you'll have an infinite loop.

Comment (mhkabir) [reply src/drivers/distance_sensor/tfmini/tfmini.cpp]: Thanks for the catch. Will fix.

**Comment by Antiheavy** [comment]
finally was able to ground test Master after this PR went it.  much better response now!

**Pull request merged**